README.md
# interactive_maps_marker for Flutter
Easy to use Animated Maps Markers with detail card. Use it for store or any place locator.
## Demo
![](demo/simple-demo.gif)

## Usage
Add this package to your `pubspec.yaml` in `dependencies:` section
```yaml
dependencies: 
  interactive_maps_marker: ^0.0.2
```
This package depends upon ```google_maps_flutter``` so first setup this by following [This Guide](https://pub.dev/packages/google_maps_flutter)

Update dependencies 
```
flutter pub get
```
You can now add a ``` InteractiveMapsMarker ``` widget to your widget tree.


### Simple Usage
In your widget import the package and use ```InteractiveMapsMarker``` Widget

**Example**
```dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';
import 'package:interactive_maps_marker/interactive_maps_marker.dart';

class ExplorePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {

    return InteractiveMapsMarker(
      items: <MarkerItem>[
        MarkerItem(id: 1, latitude: 31.4673274, longitude: 74.2637687),
        MarkerItem(id: 2, latitude: 31.4718461, longitude: 74.3531591),
      ],
      center: LatLng(31.4906504, 74.319872),
      itemContent: (context, index) {
        return Text("Current Item $index");
      },
    );

  }
}
```

### Advanced Usage
Coming Soon

### Markers data from a remote server?
Use this widget inside a stateful widget and update the markers list state with new data. An example can be found in ```stateful_example.dart``` file in example project.

## Customise with parameters
You can customise it by passing various parameter values. Details about all parameters is as follows.

Parameter | Type                              | Default Value | Description
--------- |-----------------------------------| ------------- | -----------
items | ``` List<MarkerItem> ```          | none | List of Markers with type of MarkerItem. This parameter is required and cannot be null.
itemContent | ``` IndexedWidgetBuilder ```      | none | This is builder function which will receive ``` context ``` and ``` index ```. You must return a Widget which will show on a pre designed container. This is exactly like you use ListView Builder. Not applicable when using ``` itemBuilder ```
center | ``` LatLng ```                    | ``` LatLng(0.0, 0.0) ``` | Center point for initial map display.
zoom | ``` double ```                    | 12.0 | Default zoom value for initial map screen.
itemHeight | ``` double ```                    | 116.0 | Height of your detail item. Not applicable when using ``` itemBuilder ```
itemPadding | ``` EdgeInsetsGeometry ```        | ``` EdgeInsets.only(bottom: 80.0) ``` | Padding for item detail card. Mainly used for bottom padding.
itemPadding | ``` Alignment ```                 | ``` Alignment.bottomCenter ``` | Alignment for content slider.
itemBuilder | ``` IndexedWidgetBuilder ```      | null | If you don't want default container and want to build the bottom widget yourself, you can use this builder instead of itemContent and have full control over UI. This is builder function which will receive ``` context ``` and ``` index ```. You must return a Widget.
controller | ``` InteractiveMapsController ``` | null | A controller if you want to add the markers later and then call `rebuild()` or change the index of marker to animate with your action via `setCurrentIndex(int index)`.


## Meta

Atiq Samtia– [@AtiqSamtia](https://twitter.com/atiqsamtia) – me@atiqsamtia.com

Distributed under the MIT license.

[https://github.com/atiqsamtia/interactive_maps_marker_flutter](https://github.com/atiqsamtia/interactive_maps_marker_flutter)

## Credits
Inspired by amazing work from [Mohamed Douzane](https://github.com/moho12n)

[Github Repo](https://github.com/moho12n/animateMarkersInFlutter)

[Medium Post](https://medium.com/swlh/animating-markers-in-flutter-20483ee5df49)

## Contributing

1. Fork it (<https://github.com/atiqsamtia/interactive_maps_marker_flutter/fork>)
2. Create your feature branch (`git checkout -b feature/fooBar`)
3. Commit your changes (`git commit -am 'Add some fooBar'`)
4. Push to the branch (`git push origin feature/fooBar`)
5. Create a new Pull Request

CHANGELOG.md
## [0.0.10] - (July 14, 2023)

* FIX: markers not updating when whole dataset is replaced

## [0.0.9] - (May 23, 2023)

* Add `zoomKeepOnTap` parameter to keep remember zoom level on item when tapped.

## [0.0.8] - (May 04, 2023)

* Add `onLastItem` callback to notify when last item is reached.
* Add `zoomFocus` parameter to focus on item when tapped.

## [0.0.7] - (May 04, 2023)

* FIX `reset` method for better predictability

## [0.0.6] - (May 04, 2023)

* add `reset` method to `InteractiveMapsMarkerController`

## [0.0.5] - (April 09, 2023)

* Add `InteractiveMapsController` for communicating after the map is loaded.

## [0.0.4] - (April 08, 2023)

* Flutter 3 & null safety support

## [0.0.3] - (January 08, 2021)

* Alignment bug fix

## [0.0.2] - (January 08, 2021)

* Update Google Maps dependency
* New option for alignment of slider

## [0.0.1] - (July 20, 2020)

* Initial release


example/lib/simple_usage.dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';
import 'package:interactive_maps_marker/interactive_maps_marker.dart';

class SimpleUsage extends StatelessWidget {
  final List<MarkerItem> markers = []
    ..add(MarkerItem(id: 1, latitude: 31.4673274, longitude: 74.2637687))
    ..add(MarkerItem(id: 2, latitude: 31.4718461, longitude: 74.3531591))
    ..add(MarkerItem(id: 3, latitude: 31.5325107, longitude: 74.3610325))
    ..add(MarkerItem(id: 4, latitude: 31.4668809, longitude: 74.31354));

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Simple Usage'),
      ),
      body: InteractiveMapsMarker(
        items: markers,
        center: LatLng(31.4906504, 74.319872),
        itemContent: (context, index) {
          MarkerItem item = markers[index];
          return BottomTile(item: item);
        },
      ),
    );
  }
}

class BottomTile extends StatelessWidget {
  const BottomTile({required this.item});

  final MarkerItem item;

  @override
  Widget build(BuildContext context) {
    return Container(
      child: Row(
        children: <Widget>[
          Container(width: 120.0, color: Colors.red),
          Expanded(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  Text("Store Name ${item.id}", style: Theme.of(context).textTheme.headline5),
                  Text("${item.latitude} , ${item.longitude}", style: Theme.of(context).textTheme.caption),
                  stars(),
                  Expanded(
                    child: Text('Cras et ante metus. Vivamus dignissim augue sit amet nisi volutpat, vitae tincidunt lacus accumsan. '),
                  )
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }

  Row stars() {
    return Row(
      children: <Widget>[
        Icon(Icons.star, color: Colors.orangeAccent),
        Icon(Icons.star, color: Colors.orangeAccent),
        Icon(Icons.star, color: Colors.orangeAccent),
        Icon(Icons.star_half, color: Colors.orangeAccent),
        Icon(Icons.star_border, color: Colors.orangeAccent),
        SizedBox(width: 3.0),
        Text('3.5', style: TextStyle(color: Colors.orangeAccent, fontSize: 24.0, fontWeight: FontWeight.w600))
      ],
    );
  }
}
